src/utils/detect-server.js

```diff
@@ -67,10 +67,11 @@ const serverSettings = async (devConfig, flags, projectDir, log) => {
         return null
       }
     })
-    for (const detector of detectors) {
-      const detectorResult = detector(projectDir)
-      if (detectorResult) settingsArr.push(detectorResult)
-    }
+    detectors &&
+      detectors.forEach((detector) => {
```

Comment:
How about using `map` and `filter`? `const settingsArr = detectors.map(...)`

Reply:
We'd end up having the same issue: `Cannot read property 'map' of undefined` . An alternative to this could be `(detectors || []).someArrayMethod()`

Reply:
How does this look?
```
const detectors = detectorsFiles
      ? detectorsFiles.map((det) => {
          try {
            return loadDetector(det)
          } catch (error) {
            console.error(error)
            return null
          }
        })
      : []
 ```

Reply:
I've implemented this:
```
const detectors = detectorsFiles
      ? detectorsFiles.map((det) => {
          try {
            return loadDetector(det)
          } catch (error) {
            console.error(error)
            return null
          }
        })
      : []

    const settingsArr = detectors.map((detector) => detector(projectDir)).filter((el) => el)    
```

Reply:
Removed the ternary since `detectorsFiles` comes from an internal system call (`fs.readDirSync()`) and will have an empty array if it doesn't find anything.